### PR TITLE
Add and remove connected groups

### PR DESCRIPTION
**Pull request by zak39**
This PR allows to call the API from the front-end to add one or several connected groups or remove them.

### CODE REVIEW

File: src/store/actions.js [hunk 1]
```diff
@@ -296,6 +296,39 @@ export default {
 	updateSpace(context, { space }) {
 		context.commit('updateSpace', space)
 	},
+	removeConnectedGroup(context, { spaceId, gid, name }) {
+		axios.delete(generateUrl(`/apps/workspace/spaces/${spaceId}/connected-groups/${gid}`))
+			.then((resp) => {
+			})
+			.catch((e) => {
+				console.error(e)
```
Comment (smarinier) [inline src/store/actions.js:304]: Shouldn't we notify clearly the error ?

File: src/store/actions.js [hunk 2]
```diff
@@ -296,6 +296,39 @@ export default {
...
+		.then(resp => {
+			context.commit('addConnectedGroupToWorkspace', { name, group })
+			const users = resp.data.users
+			for (const user in users) {
+				context.commit('addUserToWorkspace', { name, user: users[user] })				
+			}
+			return resp.data
+		})
+		.catch(error => {
+			console.error('Error to add connected group', error)
```
Comment (smarinier) [inline src/store/actions.js:327]: Same comment on notifications

File: lib/Controller/ConnectedGroupController.php [hunk 3]
```diff
@@ -71,13 +75,32 @@ public function addGroup(int $spaceId, string $gid): JSONResponse {
...
+
+		foreach ($group->getUsers() as $user) {
+			$users[$user->getUID()] = $this->userService->formatUser(
+				$user,
+				$this->spaceManager->get($spaceId),
+				'user'
+			);
+		};
+		
+		foreach ($users as &$user) {
```
[the remark this answers is not in the document]
Comment (zak39) [reply lib/Controller/ConnectedGroupController.php:88]: Okay, I understand your point of view. From there, I guess the $user is present in the `$group` and `SPACE-U-<ID>`.

We don't have a method to get groups from the user. Even if one exists, you will have to filter in the workspace context.

I'm thinking how to refactor this point :+1:

**Review by smarinier — COMMENTED** [review]
Still have console.error (even twice

**Comment by zak39** [comment]
> Still have console.error (even twice

Which console.error are we seeing here?

**Pull request merged**